processor/processortest/shutdown_verifier.go

```diff
@@ -44,9 +44,61 @@ func verifyTracesDoesNotProduceAfterShutdown(t *testing.T, factory processor.Fac
 	assert.EqualValues(t, generatedCount, nextSink.SpanCount())
 }
 
+func verifyLogsDoesNotProduceAfterShutdown(t *testing.T, factory processor.Factory, cfg component.Config) {
+	// Create a proc and output its produce to a sink.
+	nextSink := new(consumertest.LogsSink)
+	proc, err := factory.CreateLogsProcessor(context.Background(), NewNopCreateSettings(), cfg, nextSink)
+	if err != nil {
+		if errors.Is(err, component.ErrDataTypeIsNotSupported) {
+			return
+		}
+		require.NoError(t, err)
+	}
+	assert.NoError(t, proc.Start(context.Background(), componenttest.NewNopHost()))
+
+	// Send some logs to the proc.
+	const generatedCount = 10
+	for i := 0; i < generatedCount; i++ {
+		require.NoError(t, proc.ConsumeLogs(context.Background(), testdata.GenerateLogs(1)))
+	}
+
+	// Now shutdown the proc.
+	assert.NoError(t, proc.Shutdown(context.Background()))
+
+	// The Shutdown() is done. It means the proc must have sent everything we
+	// gave it to the next sink.
+	assert.EqualValues(t, generatedCount, nextSink.LogRecordCount())
+}
+
+func verifyMetricsDoesNotProduceAfterShutdown(t *testing.T, factory processor.Factory, cfg component.Config) {
+	// Create a proc and output its produce to a sink.
+	nextSink := new(consumertest.MetricsSink)
+	proc, err := factory.CreateMetricsProcessor(context.Background(), NewNopCreateSettings(), cfg, nextSink)
+	if err != nil {
+		if errors.Is(err, component.ErrDataTypeIsNotSupported) {
+			return
+		}
+		require.NoError(t, err)
+	}
+	assert.NoError(t, proc.Start(context.Background(), componenttest.NewNopHost()))
+
+	// Send some metrics to the proc. testdata.GenerateMetrics creates metrics with 2 data points each.
+	const generatedCount = 10
+	for i := 0; i < generatedCount; i++ {
+		require.NoError(t, proc.ConsumeMetrics(context.Background(), testdata.GenerateMetrics(1)))
+	}
+
+	// Now shutdown the proc.
+	assert.NoError(t, proc.Shutdown(context.Background()))
+
+	// The Shutdown() is done. It means the proc must have sent everything we
+	// gave it to the next sink.
+	assert.EqualValues(t, generatedCount*2, nextSink.DataPointCount())
+}
+
 // VerifyShutdown verifies the processor doesn't produce telemetry data after shutdown.
 func VerifyShutdown(t *testing.T, factory processor.Factory, cfg component.Config) {
```

Comment:
You can write a test, that also can be used as an example for the nop processor.

Reply:
sounds good

Reply:
test won't pass as nopprocessor doesn't pass on data - I suppose that's to be the expected behavior of the nopprocessor? Not sure. In the mean time I'll work on a simple processor accepting sinks.